server.js

```diff
@@ -0,0 +1,178 @@
+
+app.use(bodyParser.json());
+app.use(express.static(__dirname + '/public'));
+app.set('port', process.env.PORT || 3000);
+
+// Station endpoints
+
+app.get('/api/v1/stations', (request, response) => {
+	database('stations').select()
+		.then(stations => response.status(200).json(stations))
+		.catch(error => response.status(500).json({
+			error: error.message
+		}));
+})
+
+app.post('/api/v1/stations', (request, response) => {
+	const station = request.body;
+
+	for(let requiredParam of [
+		'station_name', 
+		'station_phone', 
+		'street_address', 
+		'city', 
+		'state', 
+		'zip_code', 
+		'latitude', 
+		'longitude', 
+		'intersection_directions', 
+		'access_days_time'
+		]) {
+		if (!station[requiredParam]) {
+			return response.status(422).json({
+				error: `Expected format: { 
+					station_name: <String>, 
+					station_phone: <String>, 
+					street_address: <String>, 
+					city: <String>, 
+					state: <String>, 
+					zip_code: <String>, 
+					latitude: <Float>, 
+					longitude: <Float>, 
+					intersection_directions: <String>, 
+					access_days_time: <String> }.  
+
+					You're missing the ${requiredParam} property.`
+			})
+		}
+	}
+
+	database('stations').insert(station, 'id')
+		.then(stationIds => response.status(201).json({
+			id: stationIds[0],
+			message: `Project "${station.station_name}" successfully created!`
+		}))
+		.catch(error => response.status(500).json({
+			error: error.message
+		}))
+})
+
+app.get('/api/v1/stations/:station_id', (request, response) => {
+	const { station_id } = request.params
+
+	database('stations').where('station_id', station_id).select()
+		.then(stations => response.status(200).json(stations))
+		.catch(error => response.status(500).json({
+			error: error.message
+		}));
+})
+
+app.put('/api/v1/stations/:station_id', (request, response) => {
+	const newName = request.body.station_name;
+	const { station_id } = request.params
+	const station = database.where('station_id', station_id).select()
+	const oldName = station.station_name
+
+	if(!station) return response.status(404).json({ error: `station with id of ${station_id} was not found.`});
+	else if (!newName) return response.status(422).json({ error: 'No station name provided' });
+
+	database.where('station_name', oldName).update('station_name', newName)
+
+	response.status(202).json({
+		message: `Edit successful. Station with id of ${station_id} name changed from ${oldName} to ${newName}.`
+	})
+})
+
+app.delete('/api/v1/stations/:station_id', (request, response) => {
+	const { station_id } = request.params
+	database('stations').where('id', station_id).delete();
+		.then(station => response.status(200).json({
+			station,
+			message: `Station ${station_id} has been deleted.`
+		}))
+		.catch(error => response.status(500).json({
+			error: `Error deleting station: ${error.message}`
+		}))
+});
+
+// Cafe endpoints
+
+app.get('/api/v1/stations/:station_id/cafes', (request, response) => {
+	database('cafes').select()
```

Comment:
If we're doing this endpoint, the database selection should only show the cafe's where the station_id matches the station_id in the request.body